src/services/asset-discovery-service.ts

```diff
@@ -1,46 +1,38 @@
 import * as pool from 'generic-pool'
 import * as puppeteer from 'puppeteer'
+import { AssetDiscoveryConfiguration } from '../configuration/asset-discovery-configuration'
+import { DEFAULT_CONFIGURATION } from '../configuration/configuration'
 import { SnapshotOptions } from '../percy-agent-client/snapshot-options'
 import logger, {logError, profile} from '../utils/logger'
 import waitForNetworkIdle from '../utils/wait-for-network-idle'
 import PercyClientService from './percy-client-service'
 import ResponseService from './response-service'
 
-interface AssetDiscoveryOptions {
-  networkIdleTimeout?: number
-}
-
-const DEFAULT_PAGE_POOL_SIZE = process.env.PERCY_POOL_SIZE
+export const MAX_SNAPSHOT_WIDTHS: number = 10
 
-export default class AssetDiscoveryService extends PercyClientService {
+export class AssetDiscoveryService extends PercyClientService {
   responseService: ResponseService
   browser: puppeteer.Browser | null
   pagePool: pool.Pool<puppeteer.Page> | null
 
-  readonly DEFAULT_NETWORK_IDLE_TIMEOUT: number = 50 // ms
-  networkIdleTimeout: number // ms
-
-  readonly MAX_SNAPSHOT_WIDTHS: number = 10
-  readonly PAGE_POOL_SIZE_MIN: number = 2
-  readonly PAGE_POOL_SIZE_MAX: number = DEFAULT_PAGE_POOL_SIZE ? parseInt(DEFAULT_PAGE_POOL_SIZE) : 10
+  configuration: AssetDiscoveryConfiguration
 
-  // Default widths to use for asset discovery. Must match Percy service defaults.
-  readonly DEFAULT_WIDTHS: number[] = [1280, 375]
-
-  constructor(buildId: number, options: AssetDiscoveryOptions = {}) {
+  constructor(buildId: number, configuration?: AssetDiscoveryConfiguration) {
     super()
     this.responseService = new ResponseService(buildId)
-    this.networkIdleTimeout = options.networkIdleTimeout || this.DEFAULT_NETWORK_IDLE_TIMEOUT
     this.browser = null
     this.pagePool = null
+    this.configuration = configuration || DEFAULT_CONFIGURATION.agent['asset-discovery']
   }
 
   async setup() {
     profile('-> assetDiscoveryService.setup')
+
     const browser = this.browser = await this.createBrowser()
     this.pagePool = await this.createPagePool(() => {
       return this.createPage(browser)
-    }, this.PAGE_POOL_SIZE_MIN, this.PAGE_POOL_SIZE_MAX)
+    }, this.configuration['page-pool-size-min'],
```

Comment:
this is one of the key bits of this PR. We're now supplying a configurable min/max pool size.